Valigator.Core/Core/Model.cs

```diff
@@ -107,15 +110,34 @@ private static Expression CreatePropertyDescriptor(Expression modelExpression, P
 			return Expression.Lambda<Func<TModel, ValidationError[][]>>(arrayInitializer, modelParameter).Compile();
 		}
 
-		private static PropertyInfo[] GetAllProperties(Type type)
-			=> GetBaseProperties(type)
-				.Concat(GetExplicitProperties(type))
+		private static Expression CreateDataExpression(Expression modelExpression, PropertyInfo propertyInfo, TModel model)
+			=> model is ValigatorModelBase
+				? Expression.Call(
```

Comment:
If this is a `ValigatorModelBase`, we use the `GetMember` and `SetMember` calls instead of expecting it to be a property.